phones/models.py

```diff
@@ -295,7 +296,21 @@ def save(self, *args, **kwargs):
         # if this number exists for this user, this is an update call
         existing_numbers = RelayNumber.objects.filter(user=self.user)
         this_number = existing_numbers.filter(number=self.number).first()
+        update_user_profile_last_engagement = False
         if this_number and this_number.id == self.id:
+            update_user_profile_last_engagement = any(
+                [
+                    self.enabled != this_number.enabled,
+                    self.calls_forwarded != this_number.calls_forwarded,
+                    self.calls_blocked != this_number.calls_blocked,
+                    self.texts_forwarded != this_number.texts_forwarded,
+                    self.texts_blocked != this_number.texts_blocked,
+                ]
+            )
+            if update_user_profile_last_engagement:
+                profile: Profile = Profile.objects.get(user=self.user)
+                profile.last_engagement = datetime.now(UTC)
+                profile.save()
```

Comment:
I don't think the variable `profile` is needed:

```suggestion
                user.profile.last_engagement = datetime.now(UTC)
                user.profile.save()
```